### PR TITLE
Add more types and corrosponding spec tests

### PR DESCRIPTION
**Pull request by b4ldr**
Hi I have added some more custom types and created some tests as well as adding some boiler plat stuff from retrospect puppet.  you probably dont want to accept all if this but if there is anything you like i could create a pull request with just thoughs elements

### CODE REVIEW

**Comment by bastelfreak** [comment]
Hi @b4ldr, thanks for the great work! Can you please take a look at the merge conflicts?

File: .fixtures.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+fixtures:
+    symlinks:
+      tea:  "#{source_dir}"
+      test: "#{source_dir}/spec/fixtures/test"
+    repositories:
+      stdlib:
+        repo: https://github.com/puppetlabs/puppetlabs-stdlib.git
+        ref: 4.12.0
```
Comment (bastelfreak) [inline .fixtures.yml]: can you drop the ref please? We try to always test against master, to early detect bugs.

File: .puppet-lint.rc [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# Use this file to instruct puppet-lint to ignore certain checks.
```
Comment (bastelfreak) [inline .puppet-lint.rc]: can you please remove the file? We manage all the puppet-lint settings in our `Rakefile`.

File: .travis.yml [hunk 3]
```diff
@@ -1,34 +1,41 @@
 ---
```
Comment (bastelfreak) [inline .travis.yml:1]: all these changes will be overwritten by our modulesync_config:
https://github.com/voxpupuli/modulesync_config/blob/master/config_defaults.yml#L2-L23

Which results in this travis file:
https://github.com/voxpupuli/puppet-tea/blob/master/.travis.yml

Our current template allows a few customizations, let me know if any of the underlying changes are still needed.

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
+# module name
```
Comment (bastelfreak) [inline DEVELOPMENT.md]: We have a similar file in .github/CONTRIBUTING.md. Is this file still relevant? Maybe we should merge it?

File: Gemfile [hunk 5]
```diff
@@ -1,14 +1,23 @@
 source "https://rubygems.org"
```
Comment (bastelfreak) [inline Gemfile]: The whole file is managed by our modulesync_config:
https://github.com/voxpupuli/modulesync_config/blob/master/config_defaults.yml#L24-L61

which results in:
https://github.com/voxpupuli/puppet-tea/blob/master/Gemfile

are any of the changes here still needed?

File: README.markdown [hunk 6]
```diff
@@ -20,18 +20,24 @@ Installing this module will make it so that puppet [auto-loads](https://docs.pup
```
Comment (bastelfreak) [inline README.markdown:20]: can you rename it into README.md?

File: Rakefile [hunk 7]
```diff
@@ -1,63 +1,72 @@
-require 'rubygems'
-require 'bundler/setup'
-
 require 'puppetlabs_spec_helper/rake_tasks'
```
Comment (bastelfreak) [inline Rakefile:1]: same here, completely managed by modulesync:
https://github.com/voxpupuli/modulesync_config/blob/master/config_defaults.yml#L62-L71

let me know if any of the changes are needed and we will fiddle something out.

File: examples/init.pp [hunk 8]
```diff
@@ -1,9 +0,0 @@
-class tea {
```
Comment (bastelfreak) [inline examples/init.pp:1]: any reason for removing this and the other example?

File: metadata.json [hunk 9]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "puppet-tea",
-  "version": "0.2.0",
+  "version": "0.2.4",
```
Comment (bastelfreak) [inline metadata.json]: please don't adjust the version number. we will take care of it during the release process.

File: spec/spec_helper.rb [hunk 10]
```diff
@@ -0,0 +1,42 @@
+require 'puppetlabs_spec_helper/module_spec_helper'
```
Comment (bastelfreak) [inline spec/spec_helper.rb:1]: this file is also managed by modulesync. Any of the changes still needed? Prefered way for customazation is to throw the code into a spec/spec_helper_methods.rb and include it here. Example:
https://github.com/voxpupuli/puppet-zabbix/blob/master/spec/spec_helper.rb#L15
https://github.com/voxpupuli/puppet-zabbix/blob/master/.sync.yml#L4-L5
https://github.com/voxpupuli/modulesync_config/blob/master/moduleroot/spec/spec_helper.rb#L15-L17

**Review by bastelfreak — CHANGES_REQUESTED** [review]

**Comment by bastelfreak** [comment]
also if you could split this up into seperate commits that would be great, it can stay in one PR.

**Comment by b4ldr** [comment]
i have redone this in https://github.com/voxpupuli/puppet-tea/pull/15

**Pull request closed without merging**